src/databricks/labs/ucx/source_code/queries.py

```diff
@@ -0,0 +1,53 @@
+import logging
+from collections.abc import Iterable
+from dataclasses import asdict
+from datetime import datetime
+
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.service.sql import ListQueryObjectsResponseQuery
+
+
+from databricks.labs.ucx.source_code.directfs_access import DirectFsAccessCrawler, DirectFsAccessInQuery
+from databricks.labs.ucx.source_code.linters.directfs import DirectFsAccessSqlLinter
+
+logger = logging.getLogger(__name__)
+
+
+class QueryLinter:
+
+    def __init__(
+        self,
+        ws: WorkspaceClient,
+        directfs_crawler: DirectFsAccessCrawler,
+    ):
+        self._ws = ws
+        self._directfs_crawler = directfs_crawler
+
+    def collect_dfsas_from_queries(self) -> Iterable[DirectFsAccessInQuery]:
+        assessment_start = datetime.now()
+        queries_dfsas = self._collect_dfsas_from_queries()
+        assessment_end = datetime.now()
+        dfsas: list[DirectFsAccessInQuery] = []
+        for dfsa in queries_dfsas:
+            dfsas.append(
+                dfsa.replace_assessment_infos(assessment_start=assessment_start, assessment_end=assessment_end)
+            )
+        self._directfs_crawler.append(dfsas)
+        return dfsas
+
+    def _collect_dfsas_from_queries(self) -> Iterable[DirectFsAccessInQuery]:
+        queries = self._ws.queries.list()
+        for query in queries:
+            yield from self._collect_dfsas_from_query(query)
+
+    @classmethod
+    def _collect_dfsas_from_query(cls, query: ListQueryObjectsResponseQuery) -> Iterable[DirectFsAccessInQuery]:
+        if query.query_text is None:
+            return
+        linter = DirectFsAccessSqlLinter()
```

Comment:
while it's early, do you think it makes sense to add it to `GlobalContext` or some other higher level factory?

Reply:
It's stateless and not context dependent so I'd say no